libraries/node-core-library/src/JsonSchema.ts

```diff
@@ -308,6 +335,11 @@ export class JsonSchema {
       // Enable json-schema format validation
       // https://ajv.js.org/packages/ajv-formats.html
       addFormats(validator);
+      if (this._customFormats) {
+        Object.entries(this._customFormats).forEach(([name, format]) => {
+          validator.addFormat(name, { ...format, async: false });
```

Comment:
I'm curious why we would use `IJsonSchemaCustomFormat` vs the underlying type here? Is it so that we don't allow _too much_ customization in the formatting or to simplify the use case?

Reply:
I figured we wouldn't necessarily want to leak/support the entire AJV API

Reply:
Good enough reasoning for me. I'd prefer to support a limited use case and expand as needed here as well